daemon/main.go

```diff
@@ -336,9 +335,6 @@ func init() {
 	flags.String("enable-policy", endpoint.DefaultEnforcement, "Enable policy enforcement")
 	flags.BoolVar(&enableTracing,
 		"enable-tracing", false, "Enable tracing while determining policy (debugging)")
-	flags.BoolVar(&useEnvoy,
```

Comment:
What happens to current users that are running cilium with `--envoy-proxy`? I think this flag should still exist either with `(deprecated)` and we remove it on the next version or we market it as an hidden flag in viper.

Reply:
I think removal of the flag is safe pre 1.0. Are you worried about the error if the flag is still passed in?

Reply:
correct, there will be an error like:
```
$ cilium-agent --envoy-proxy
Error: unknown flag: --envoy-proxy
```

Reply:
I'm fine to keep it around for another release but we should remove it pre 1.0.

Reply:
@aanm Do we have an example of how to hide the flag somewhere?

Reply:
I'll try with ```flags.MarkHidden("envoy-proxy")```

Reply:
yes, I think the `markhidden` does the trick we are using with `cmdref`